packages/manager/apps/telecom/src/app/telecom/telephony/line/assist/rma/rma.controller.js

```diff
@@ -1,101 +1,159 @@
 import remove from 'lodash/remove';
-import set from 'lodash/set';
 
-export default /* @ngInject */ function TelecomTelephonyLineAssistRmaCtrl(
-  $stateParams,
-  $q,
-  $translate,
-  TucToast,
-  TucToastError,
-  OvhApiTelephony,
-) {
-  const self = this;
+import { TEXT_FOR_MODAL, ACTION_TYPE, RMA_NEW_TYPE } from './rma.constants';
 
-  function init() {
-    self.rmaList = null;
-    return self
-      .fetchRma()
+export default class TelecomTelephonyLineAssistRmaCtrl {
+  constructor($translate, lineAssistRmaService, TucToast, TucToastError) {
+    this.$translate = $translate;
+    this.lineAssistRmaService = lineAssistRmaService;
+    this.TucToast = TucToast;
+    this.TucToastError = TucToastError;
+  }
+
+  $onInit() {
+    this.rmaList = null;
+    this.modalPrimaryAction = null;
+    this.modalTitle = null;
+    this.modalContent = null;
+    this.modalRma = null;
+    this.isOpenModal = false;
+    this.actionDoing = false;
+
+    return this.lineAssistRmaService
+      .fetchRma(this.billingAccount, this.serviceName)
       .then((result) => {
-        self.rmaList = result;
+        this.rmaList = result;
       })
-      .catch((err) => new TucToastError(err));
+      .catch((err) => new this.TucToastError(err));
   }
 
-  self.fetchPhone = function fetchPhone() {
-    return OvhApiTelephony.Line()
-      .Phone()
-      .v6()
-      .get({
-        billingAccount: $stateParams.billingAccount,
-        serviceName: $stateParams.serviceName,
-      }).$promise;
-  };
+  modalDismiss() {
+    this.isOpenModal = false;
+    this.modalPrimaryAction = null;
+    this.modalTitle = null;
+    this.modalContent = null;
+    this.modalRma = null;
+  }
```

Comment:
**suggestion:** Can be renamed to `resetModalInfo` and use it on `$onInit` as well.

Reply:
I fixed.